flink-runtime/src/main/java/org/apache/flink/runtime/rest/RestClient.java

```diff
@@ -247,7 +249,15 @@ public void shutdown(Time timeout) {
 
 		@Override
 		protected void channelRead0(ChannelHandlerContext ctx, Object msg) {
-			if (msg instanceof FullHttpResponse) {
+			// TODO: should this check for status OK (200) and treat all other as errors?
```

Comment:
message semantics are handled later when parsing the payload, which effectively does what you're suggesting. The else branch for debugging; our rest servers always return `FullHttpResponses`.

Reply:
Thx for filling this out. In that case should I just drop this `// TODO:`?

Reply:
yes I think we can just drop it.